tests/proxyMap.test.tsx

```diff
@@ -375,3 +376,145 @@ describe('snapshot', () => {
     expect(snap2.get('key1')).toBe('val1modified')
   })
 })
+
+describe('ui updates - useSnapshot', async () => {
+  it('should update ui when calling has before and after setting and deleting a key', async () => {
+    const state = proxyMap()
+    const TestComponent = () => {
+      const snap = useSnapshot(state)
+
+      return (
+        <>
+          <p>has key: {`${snap.has('key')}`}</p>
+          <button onClick={() => state.set('key', 'value')}>set key</button>
+          <button onClick={() => state.delete('key')}>delete key</button>
+        </>
+      )
+    }
+
+    const { getByText } = render(
+      <StrictMode>
+        <TestComponent />
+      </StrictMode>,
+    )
+
+    await waitFor(() => {
+      getByText('has key: false')
+    })
+
+    fireEvent.click(getByText('set key'))
+    await waitFor(() => {
+      getByText('has key: true')
+    })
+
+    fireEvent.click(getByText('delete key'))
+    await waitFor(() => {
+      getByText('has key: false')
+    })
+  })
+
+  it('should update ui when calling has before and after settiing and deleting multiple keys', async () => {
+    const state = proxyMap()
+    const TestComponent = () => {
+      const snap = useSnapshot(state)
+
+      return (
+        <>
+          <p>has key: {`${snap.has('key')}`}</p>
+          <p>has key2: {`${snap.has('key2')}`}</p>
```

Comment:
I think we should have a test that deletes the first key only. (Did I miss it somewhere?)

Reply:
I'll add it

Reply:
I went ahead and added another test that tests deleting the first key as well as deleting only the second (or last) key. Couldn't hurt.